mthamayor/auto-mart
# #166394429 Users can reset their password

#### What does this PR do?
Users can reset their password

#### Description of Task to be completed?
Create API endpoint POST /auth/forgot
Create API endpoint POST /auth/reset

Unit tests should be written for the endpoints
The endpoints should be tested on Postman

#### How should this be manually tested?
After cloning the repo

Start the development server with
`npm run dev`
After signing up,
Access the forgot password endpoint
http://localhost:3000/api/v1/auth/forgot

add the following fields to the form body
email: <your email>

A link would be sent to the email provided.

copy the token sent to the email and access the reset endpoint
http://localhost:3000/api/v1/auth/reset

add the following fields to the form body and hit send
token: <token from email>
newPassword: <Your new password>

Hit send on postman
You should receive a response 
```
{
    "status": 200,
    "data": {
         "data": "Password successfully changed"
    }
}
```
#### Any background context you want to provide?
This feature is only available in the production environment

#### What are the relevant pivotal tracker stories?
https://www.pivotaltracker.com/story/show/166394429
